core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -328,6 +339,13 @@ public WorkerNetAddress location() {
 
   @Override
   public boolean isLocal() {
-    return mLocal;
+    return CommonUtils.isLocalHost(mAddress);
```

Comment:
I don't think so. `IsLocal` is from `Locatable` which is based on `InetSocketAddress`.

For example `IsLocal` can return true when the file is not in Alluxio and the local worker reads from UFS.